internal/utils/index/extendable.go

```diff
@@ -0,0 +1,123 @@
+package index
```

Comment:
Could we add more comments to tell what `OptionsForExtendables` is used for?

Reply:
To make the pattern consistent, I split `OptionsForExtendables` into `OptionsForDataPlane`, `OptionsForGatewayConfiguration`, and `OptionsForControlPlane`. PTAL https://github.com/Kong/gateway-operator/compare/a6d4244a85e207c6fb5cd491a0cc424be98db329..459d80229b863ab93be00260dac19d7185f66b15